### PR TITLE
Orthogonal

### PR DESCRIPTION
**Pull request by davidpiuva**
Refactoring code related to the projection matrix and deprecated draw calls. Fixed an old bug with orthogonal clip planes.

### CODE REVIEW

**Comment by davidpiuva** [comment]
@emilwestergren Peer review?

File: source/draw/gpu/opengl/OpenGLCanvas.ooc [hunk 1]
```diff
@@ -24,16 +24,19 @@ OpenGLCanvas: class extends OpenGLSurface {
 		viewport := (drawState viewport hasZeroArea) ? IntBox2D new(this size) : drawState viewport
 		this context backend setViewport(viewport)
 		focalLengthPerWidth := drawState getFocalLengthNormalized()
-		aspectRatio := (this size x as Float) / (this size y as Float)
+		focalLengthPerHeight := focalLengthPerWidth * (this size x as Float) / (this size y as Float)
 		gpuMap view = _toLocal * drawState getTransformNormalized() normalizedToReference(this size) * _toLocal
+		flipY := -(this _coordinateTransform e as Float)
 		if (focalLengthPerWidth > 0.0f) {
+			nearPlane := 0.01f
+			farPlane := 10000.0f
```
Comment (davidpiuva) [inline source/draw/gpu/opengl/OpenGLCanvas.ooc]: Near plane is less to allow normalization.

File: source/draw/gpu/opengl/OpenGLCanvas.ooc [hunk 2]
```diff
@@ -24,16 +24,19 @@ OpenGLCanvas: class extends OpenGLSurface {
 		viewport := (drawState viewport hasZeroArea) ? IntBox2D new(this size) : drawState viewport
 		this context backend setViewport(viewport)
 		focalLengthPerWidth := drawState getFocalLengthNormalized()
-		aspectRatio := (this size x as Float) / (this size y as Float)
+		focalLengthPerHeight := focalLengthPerWidth * (this size x as Float) / (this size y as Float)
 		gpuMap view = _toLocal * drawState getTransformNormalized() normalizedToReference(this size) * _toLocal
+		flipY := -(this _coordinateTransform e as Float)
```
Comment (davidpiuva) [inline source/draw/gpu/opengl/OpenGLCanvas.ooc]: The expression was too messy and confusing.

File: source/draw/gpu/opengl/OpenGLCanvas.ooc [hunk 3]
```diff
@@ -24,16 +24,19 @@ OpenGLCanvas: class extends OpenGLSurface {
...
-			f := -(this _coordinateTransform e as Float) * 2.0f * focalLengthPerWidth * aspectRatio
-			k := (this _farPlane + this _nearPlane) / (this _farPlane - this _nearPlane)
-			o := 2.0f * this _farPlane * this _nearPlane / (this _farPlane - this _nearPlane)
+			f := flipY * 2.0f * focalLengthPerHeight
+			k := (farPlane + nearPlane) / (farPlane - nearPlane)
+			o := 2.0f * farPlane * nearPlane / (farPlane - nearPlane)
 			gpuMap projection = FloatTransform3D new(a, 0.0f, 0.0f, 0.0f, 0.0f, f, 0.0f, 0.0f, 0.0f, 0.0f, k, -1.0f, 0.0f, 0.0f, o, 0.0f)
 		} else
-			gpuMap projection = FloatTransform3D createScaling(2.0f / this size x, -(this _coordinateTransform e as Float) * 2.0f / this size y, 1.0f)
+			gpuMap projection = FloatTransform3D createScaling(2.0f / this size x, flipY * 2.0f / this size y, 0.0f)
```
Comment (davidpiuva) [inline source/draw/gpu/opengl/OpenGLCanvas.ooc]: Orthogonal vertices are projected against the Z = 0 plane to avoid any depth clipping.

**Comment by marcusnaslund** [comment]
You will be removing the deprecated parts pretty soon right?

**Comment by davidpiuva** [comment]
Yes. Only one thing remains that uses them.

**Comment by marcusnaslund** [comment]
Excellent.

**Pull request merged**